docs: Reword OTel Engine pages (#6781)
grafana/alloy: docs/sources/introduction/otel_alloy.md

Author: I'm trying to encourage people to try OTel Engine, but I'll need to mention somewhere that it's still experimental? But it's experimental in the sense that the way it's exposed in Alloy may change. A lot of the features it exposes are actually quite stable upstream.

Reviewer: We have a OTel engine-specific shared text block for the experimental status of the OTel Engine... we could tweak that as needed and it'll only impact the doc pages that reference the OTel engine.

Author: Thanks! Once OTel Engine goes GA, we can tweak the wording to not mention "experimental"... for now I'll just leave it as it is.

Reviewer: ```suggestion
It lets you run {{< param "PRODUCT_NAME" >}} with the OpenTelemetry Collector while retaining access to {{< param "PRODUCT_NAME" >}} features and integrations through the {{< param "PRODUCT_NAME" >}} Engine extension.
```

Don't we need the extension in this case?

Author: Yes, that's true.

Translating Collector configuration into another format can add friction, and the rapid pace of upstream development makes translation difficult to keep current.

The {{< param "OTEL_ENGINE" >}} addresses this by running the upstream Collector runtime directly from the {{< param "PRODUCT_NAME" >}} executable.
You can bring an existing Collector configuration to {{< param "PRODUCT_NAME" >}}, use familiar Collector tooling, and choose from the [components bundled with the `otel` command](../../reference/cli/otel/#included-components).

The {{< param "OTEL_ENGINE" >}} also gives Grafana a standards-native foundation for extending {{< param "PRODUCT_NAME" >}}.
Grafana is committed to providing a first-class OpenTelemetry collection experience as this experimental engine matures.

## How the engines fit together

The following tree shows how the engines and the {{< param "PRODUCT_NAME" >}} Engine extension fit inside the {{< param "PRODUCT_NAME" >}} executable:

```text
Grafana Alloy executable
├── alloy run
│   └── Default Engine
└── alloy otel
    └── OTel Engine
        └── alloyengine extension (optional)
            └── Default Engine pipeline

> ```

Author: It'd be nice to have a Mermaid diagram for this. I suppose [TreeView](https://mermaid.ai/open-source/syntax/treeView.html) is the best, but I'm not sure if the Grafana website supports it.

Author: This what it'd look like:

<img width="1096" height="557" alt="Image" src="https://github.com/user-attachments/assets/20b6a93a-a5bd-42b3-9d4a-98a2a4902c01" />

This is a picture that I'd handcrafted a while back:

<img width="1009" height="371" alt="Image" src="https://github.com/user-attachments/assets/4d3b9dfe-4b25-40c7-bb75-68ab734ebf43" />

Author: I'll remove the diagram from the doc for now.

{{< docs/shared lookup="stability/experimental_otel.md" source="alloy" version="<ALLOY_VERSION>" >}}

> ## Why the OpenTelemetry Engine exists

Reviewer: ```suggestion
## Why the {{% param "OTEL_ENGINE" %}} exists
```

> OpenTelemetry Collector users expect a distribution to accept standard Collector YAML configuration and command-line arguments.

Reviewer: This is an assumption we should probably avoid stating. 

```suggestion
Standard OpenTelemetry Collector pipelines use YAML configuration, but {{< param "PRODUCT_NAME" >}} components require translating that configuration into {{< param "PRODUCT_NAME" >}} syntax.
```

> Translating Collector configuration into another format can add friction, and the rapid pace of upstream development makes translation difficult to keep current.

Reviewer: This is the negative side. We could rephrase it positive like....

```suggestion
The {{< param "OTEL_ENGINE" >}} runs the Collector runtime directly, so you can use Collector YAML configurations without translation.
```

The {{< param "OTEL_ENGINE" >}} addresses this by running the upstream Collector runtime directly from the {{< param "PRODUCT_NAME" >}} executable.

> You can bring an existing Collector configuration to {{< param "PRODUCT_NAME" >}}, use familiar Collector tooling, and choose from the [components bundled with the `otel` command](../../reference/cli/otel/#included-components).

Reviewer: ```suggestion
You can bring existing Collector configurations to {{< param "PRODUCT_NAME" >}}, use familiar Collector tooling, and choose from the [components bundled with the `otel` command](../../reference/cli/otel/#included-components).
```

The {{< param "OTEL_ENGINE" >}} also gives Grafana a standards-native foundation for extending {{< param "PRODUCT_NAME" >}}.
Grafana is committed to providing a first-class OpenTelemetry collection experience as this experimental engine matures.

## How the engines fit together

> The following tree shows how the engines and the {{< param "PRODUCT_NAME" >}} Engine extension fit inside the {{< param "PRODUCT_NAME" >}} executable:

Reviewer: The diagram names bits of the OTel Engine that haven't been defined. I'm missing the answer to the implied question from the H2 for this section.

```suggestion
Both engines are built into the same {{< param "PRODUCT_NAME" >}} binary: `alloy run` starts the {{< param "DEFAULT_ENGINE" >}}, and `alloy otel` starts the {{< param "OTEL_ENGINE" >}}.
The optional {{< param "PRODUCT_NAME" >}} Engine extension lets you run a {{< param "DEFAULT_ENGINE" >}} pipeline inside the {{< param "OTEL_ENGINE" >}}, in the same process.

The following diagram shows how the engines and the extension fit inside the {{< param "PRODUCT_NAME" >}} executable:
```
